src/Microsoft.AspNet.Server.Kestrel/Infrastructure/KestrelThread.cs

```diff
@@ -120,6 +126,18 @@ private void OnStopRude(object obj)
                 IntPtr.Zero);
         }
 
+        private void OnStopForceUnreference(object obj)
+        {
+            _engine.Libuv.walk(
+                _loop,
+                (ptr, arg) =>
+                {
+                    var handle = UvMemory.FromIntPtr<UvHandle>(ptr);
+                    _loop.Libuv.unref(handle);
```

Comment:
For example `_post.Unreference()` followed by `if (handle != _post)` in the harsher stop, suggests a similar encounter of this previously.